drivers/watchdog/wdt_nrfx_wdt.c

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright (c) 2018, Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <nrfx_wdt.h>
+#include <nrf_clock.h>
+#include <clock_control.h>
+#include <watchdog.h>
+
+#define SYS_LOG_DOMAIN "WDT NRF"
+#define SYS_LOG_LEVEL CONFIG_SYS_LOG_WDT_LEVEL
+#include <logging/sys_log.h>
+
+#define NRF_WDT_CHANNELS_NUMBER 8
+
+static struct device DEVICE_NAME_GET(wdt_nrf);
+
+/* There are eight reload registers (channels) in Nordic's WDT. */
+static nrfx_wdt_channel_id m_ch[NRF_WDT_CHANNELS_NUMBER];
+
+/* Each activated and not reloaded channel can generate watchdog interrupt.
+ * Array m_cb provides storing callbacks for each channel.
+ */
+static wdt_callback_t      m_cb[NRF_WDT_CHANNELS_NUMBER];
+
+/*
+ * The m_alloc_ch_num variable stores number of currently allocated
+ * and activated watchdog channels.
+ */
+static u32_t               m_alloc_ch_num;
+
+
+static void wdt_event_handler(void)
+{
+	u32_t i;
+
+	for (i = 0; i < m_alloc_ch_num; ++i) {
+		if (nrf_wdt_request_status(m_ch[i])) {
+			if (m_cb[i]) {
+				m_cb[i](DEVICE_GET(wdt_nrf), &m_ch[i]);
+			}
+		}
+	}
+}
+
+static void wdt_nrf_isr(struct device *dev)
+{
+	ARG_UNUSED(dev);
+	/* We need to implement our own interrupt handler until nrfx one will
+	 * be fixed.
+	 * Clearing events also clears the reload register and has to be done
+	 * AFTER event handler.
+	 */
+	if (nrf_wdt_int_enable_check(NRF_WDT_INT_TIMEOUT_MASK)) {
+		wdt_event_handler();
+		nrf_wdt_event_clear(NRF_WDT_EVENT_TIMEOUT);
+	}
+}
+
+static int wdt_nrf_setup(struct device *dev, u8_t options)
+{
+	nrfx_wdt_config_t config;
+	nrfx_err_t err_code;
+
+	/* Set default value according to default value in MCU. */
+	config.behaviour = NRF_WDT_BEHAVIOUR_RUN_SLEEP;
```

Comment:
This should be `NRF_WDT_BEHAVIOUR_RUN_SLEEP_HALT`, otherwise `WDT_OPT_PAUSE_HALTED_BY_DBG` won't be handled correctly.
And perhaps it would be better to keep this line together with others related to the `config` structure?

Reply:
By default watchdog is running when CPU sleeps. I think next lines can be modified to cover all cases. I'll look at it

Reply:
Ok I got it. Changed.